src/js/app.js

```diff
@@ -1,16 +1,115 @@
 const app = angular.module('myApp', []);
 
 app.controller('MainController', ['$scope', ($scope) => {
-    $scope.message = 'We are set to build our inverted index project';
-    $scope.fileList = ['File 1', 'File2'];
-    $scope.table1IsVisible = false;
-    $scope.table2IsVisible = false;
-    $scope.showIndexTable = () => {
-        $scope.table1IsVisible = true;
-        $scope.table2IsVisible = false;
+    const invertedIndex = new InvertedIndex();
+
+    $scope.filesArray = [];
+    $scope.fileNamesArray = [];
+    $scope.file = [];
+    $scope.filesIndexCreated = [];
+    $scope.showIndexTable = false;
+    $scope.showSearchTable = false;
+    $scope.tableName = '';
+
+    /**
+     * upload single or multiple files
+     * @returns {String} success or failure message
+     */
+    $scope.uploadFile = () => {
+        const books = Array.from(document.getElementById('json-files').files);
+        // const readFile = new FileReader();
+        const booksCount = books.length;
+        if (booksCount > 0) {
+            const promise = Promise.resolve();
+            books.map(book => promise.then(() => $scope.bookReader(book)));
+            document.getElementById('json-files').value = null;
+        } else {
+            toastr.warning('No file selected', 'Warning');
+        }
     };
-    $scope.showSearchTable = () => {
-        $scope.table1IsVisible = false;
-        $scope.table2IsVisible = true;
+
+    /**
+     * read a book and store
+     * @param {object} book
+     * @returns {Promise} returns promise containing file(s) uploaded
+     */
+    $scope.bookReader = (book) => {
+        return new Promise((resolve, reject) => {
+            if (book.name.toLowerCase().match(/\.json$/)) {
+                const readFile = new FileReader();
+                readFile.onload = (e) => {
```

Comment:
Is it possible to have the body of this closure in a separate and well-named function that will just be called here? That might increase the readability of the `bookReader` function.